crt_portal/cts_forms/views.py

```diff
@@ -115,61 +115,81 @@ def IndexView(request):
+    crimes = {
+        'physical_harm': False,
+        'trafficking': False
+    }
 
-        p_class_list = format_protected_class(
-            report.protected_class.all().order_by('form_order'),
-            report.other_class,
-        )
+    for crime in report.hatecrimes_trafficking.all():
+        for choice in HATE_CRIMES_TRAFFICKING_MODEL_CHOICES:
+            if crime.hatecrimes_trafficking_option == choice[1]:
+                crimes[choice[0]] = True
+
+    p_class_list = format_protected_class(
+        report.protected_class.all().order_by('form_order'),
+        report.other_class,
+    )
+
+    output = {
+        'actions': ComplaintActions(initial={
+            'assigned_section': report.assigned_section,
+            'status': report.status
+        }),
+        'comments': CommentActions(),
+        'activity_stream': report.target_actions.all(),
+        'crimes': crimes,
+        'data': report,
+        'p_class_list': p_class_list,
+        'primary_complaint': primary_complaint,
+        'return_url_args': request.GET.get('next', ''),
+    }
+
+    return output
 
-        output = {
-            'actions': ComplaintActions(initial={
-                'assigned_section': report.assigned_section,
-                'status': report.status
-            }),
-            'activity_stream': report.target_actions.all(),
-            'crimes': crimes,
-            'data': report,
-            'p_class_list': p_class_list,
-            'primary_complaint': primary_complaint,
-            'return_url_args': request.GET.get('next', ''),
-        }
-
-        return output
 
+class ShowView(LoginRequiredMixin, View):
     def get(self, request, id):
-        output = self.__serialize_data(request, id)
+        report = get_object_or_404(Report, pk=id)
+        output = serialize_data(report, request, id)
 
         return render(request, 'forms/complaint_view/show/index.html', output)
 
     def post(self, request, id):
-        report = get_object_or_404(Report, id=id)
-
-        form = ComplaintActions(request.POST, instance=report)
-        if form.is_valid() and form.has_changed():
-            form.update_activity_stream(request.user)
-            form.save()
+        report = get_object_or_404(Report, pk=id)
+        action_form = ComplaintActions(request.POST, instance=report)
+        if action_form.is_valid() and action_form.has_changed():
+            action_form.update_activity_stream(request.user)
+            action_form.save()
 
-        output = self.__serialize_data(request, id)
+        output = serialize_data(report, request, id)
         output.update({
             'return_url_args': request.POST.get('next', ''),
         })
 
         return render(self.request, 'forms/complaint_view/show/index.html', output)
 
 
+@login_required
+def SaveComment(request, report_id):
+    """Can be used for saving comments or summaries for a report"""
+    if request.method == 'POST':
```

Comment:
We could leverage a form here to validate the input and contain the save/creation logic a bit. As-is this allows for empty comments to be submitted